### PR TITLE
Fix github auth again

### LINKED ISSUE #2728: Crash in modified GitHub auth
Are you experiencing an issue with...

- [x] [shields.io](https://shields.io/#/)

:beetle: **Description**

Post deploy of #1205:

```
Error: Vendor hook error:,Error: usesRemaining must be an integer
    at Token.update (/home/m/shields/lib/token-pool.js:65:13)
    at GithubApiProvider.updateToken (/home/m/shields/services/github/github-api-provider.js:57:11)
    at request (/home/m/shields/services/github/github-api-provider.js:109:16)
    at Request.request [as _callback] (/home/m/shields/lib/request-handler.js:182:9)
    at Request.self.callback (/home/m/shields/node_modules/request/request.js:185:22)
    at Request.emit (events.js:160:13)
    at Request.<anonymous> (/home/m/shields/node_modules/request/request.js:1161:10)
    at Request.emit (events.js:160:13)
    at IncomingMessage.<anonymous> (/home/m/shields/node_modules/request/request.js:1083:12)
    at Object.onceWrapper (events.js:255:19)
    at IncomingMessage.emit (events.js:165:20)
    at endReadableNT (_stream_readable.js:1101:12)
    at process._tickCallback (internal/process/next_tick.js:152:19)
```

### PR DESCRIPTION
**Pull request by paulmelnikow**
This code isn't being run during tests, though let's fix that later as part of #2733. Specifically:

> However _the pool itself_ could be used all the time; there's not a big advantage in turning that off when it doesn't need to be used.

Fix #2728

### CODE REVIEW

**Comment by calebcartwright** [comment]
The CI test failures do not look related to this at all.

**Comment by paulmelnikow** [comment]
I can't reproduce the search test failure locally so going to turn it off.

File: services/github/github-api-provider.spec.js [hunk 1]
```diff
@@ -77,11 +77,12 @@ describe('Github API provider', function() {
     it('should update the token with the expected values', function(done) {
       provider.request(mockRequest, '/foo', {}, (err, res, buffer) => {
         expect(err).to.equal(null)
-        const expectedUsesRemaining = remaining - reserveFraction * rateLimit
-        expect(
-          mockStandardToken.update.withArgs(expectedUsesRemaining, nextReset)
-            .calledOnce
-        ).to.be.true
+        const expectedUsesRemaining =
+          remaining - Math.ceil(reserveFraction * rateLimit)
```
Comment (calebcartwright) [inline services/github/github-api-provider.spec.js]: 👍

**Comment by calebcartwright** [comment]
🤞

**Pull request merged**